app/models/mixins/retirement_mixin.rb

```diff
@@ -10,7 +10,7 @@ module RetirementMixin
   end
 
   module ClassMethods
-    def make_retire_request(*src_ids, requester)
+    def make_retire_request(*src_ids, requester = User.find_by(:userid => 'admin'))
```

Comment:
Why 'admin' here and `User.current_user.userid` everywhere else?

Reply:
Because we shouldn't be doing retirement based on current_user, it should be from the evm_owner, and if not one of those, admin. Otherwise the system retirement can run with different users on a retry.

Reply:
agreed

```ruby
    def make_retire_request(*src_ids, requester = User.current_user)
```

Reply:
I don't want current_user here _at all_. The [place we use it](https://github.com/ManageIQ/manageiq/pull/17951/files#diff-d286dd33aa15fb98e02d16598ad92899R10) is from the UI where it makes sense that we want the user who clicked on retirement. It should not be in make_retire_request.